### PR TITLE
Duplo 16117 Nodepool List Datasource Fix

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Fixed listing of Nodepools 
## Summary of changes

This PR does the following:

- updated document
- Fixed State not getting set for data source of node pools datasource

## Testing performed

- [ ] Using unit tests
- [ ✓] Manually, on my local system
- [ ✓] Manually, on a remote test system

## Describe any breaking changes

- ...

___

## **Type**
bug_fix, enhancement, documentation

___

## **Description**
- Enhanced logging and corrected data handling in GCP Node Pools data source.
- Fixed `cgroup_mode` field type and state setting in GCP Node Pools and K8 Node Pool resource.
- Updated documentation and examples for GCP Node Pools and SQL Database Instances.
- Documented recent fixes and enhancements in the CHANGELOG.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>data_source_duplo_gcp_node_pools.go</strong><dd><code>Enhancements and Fixes in GCP Node Pools Data Source</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplocloud/data_source_duplo_gcp_node_pools.go

<li>Added <code>fmt</code> import for enhanced logging.<br> <li> Changed <code>cgroup_mode</code> field type from <code>TypeString</code> to <code>TypeList</code> to <br>correctly handle multiple values.<br> <li> Updated the <code>dataSourceGCPNodePoolList</code> function to set the <code>node_pools</code> <br>state with enhanced logging.<br> <li> Modified <code>setGCPNodePoolStateFieldList</code> function to return a map for <br>node pool state setting.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/484/files#diff-285fb8b911ee2c3a8d8a815331606e4bd2d7ed57c16824ef0583458c4edb7b46">+9/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>data-source.tf</strong><dd><code>Example Update for GCP Node Pools Data Source</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/data-sources/duplocloud_gcp_node_pools/data-source.tf

<li>Updated data source name in the example from <code>pool</code> to <code>app</code>.<br> <li> Adjusted output variable to match the updated data source name.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/484/files#diff-9f1a6ae2c2088befdb15ee89dba1042d8e3273f61ee4d13c218faddc567879eb">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Bug_fix
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>resource_duplo_gcp_k8_node_pool.go</strong><dd><code>Fixes in GCP K8 Node Pool Resource Configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplocloud/resource_duplo_gcp_k8_node_pool.go

<li>Adjusted <code>gcpNodePoolLoggingConfigToState</code> to correctly map <br><code>variant_config</code>.<br> <li> Updated <code>gcpNodePoolLinuxConfigToState</code> to handle <code>cgroup_mode</code> as a list <br>and correct <code>sysctls</code> key.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/484/files#diff-bc0cc4a27038eb399598c526fd7148bd7cd9349310b06235a990a5f2cac1eaf9">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Updated CHANGELOG for Node Pools and SQL Database Instances Fixes</code></dd></summary>
<hr>
      
CHANGELOG.md

<li>Documented fixes and enhancements related to GCP Node Pools data <br>source.<br> <li> Updated documentation for GCP SQL Database Instances.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/484/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed">+14/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>gcp_node_pools.md</strong><dd><code>Documentation Update for GCP Node Pools Data Source</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
docs/data-sources/gcp_node_pools.md

<li>Updated example usage to reflect changes in data source naming.<br> <li> Corrected documentation to match the new <code>cgroup_mode</code> data type as a <br>list.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/484/files#diff-aa56a3663198f893bc2495cf99a881cc73513cb1fb4219d80ef7c5bbc572f237">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>data-source.tf</strong><dd><code>Example Correction for GCP SQL Database Instances</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/data-sources/duplocloud_gcp_sqls/data-source.tf

- Corrected output variable name from `sqls` to `databases`.

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/484/files#diff-39978ab10934b71184364363d119915d9def083f9c4e76b1aa04911d22494f4d">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by Duplo-nikv** [comment]
Also fixed datasource for sqls and updated document

File: duplocloud/data_source_duplo_gcp_node_pools.go [hunk 1]
```diff
@@ -382,7 +388,7 @@ func dataSourceGCPNodePoolList(ctx context.Context, d *schema.ResourceData, m in
 
 func setGCPNodePoolStateFieldList(duplo *duplosdk.DuploGCPK8NodePool) map[string]interface{} {
 	// Set simple fields first.
-	return map[string]interface{}{
+	mp := map[string]interface{}{
```
Comment (komalkadam-dc) [inline duplocloud/data_source_duplo_gcp_node_pools.go:391]: do we need this declaration, we can directly return it

**Review by komalkadam-dc — CHANGES_REQUESTED** [review]

File: duplocloud/data_source_duplo_gcp_node_pools.go (hunk 1 above)
Comment (Duplo-nikv) [reply duplocloud/data_source_duplo_gcp_node_pools.go:391]: yes its better to do so, to avoid nil pointer exception

**Review by komalkadam-dc — DISMISSED** [review]

File: duplocloud/data_source_duplo_gcp_node_pools.go [hunk 2]
```diff
@@ -370,12 +382,14 @@ func dataSourceGCPNodePoolList(ctx context.Context, d *schema.ResourceData, m in
...
 	// Set simple fields first.
 	d.SetId(tenantID)
+
 	list := make([]map[string]interface{}, 0, len(*duploList))
 
 	for _, duplo := range *duploList {
 		nodepool := setGCPNodePoolStateFieldList(&duplo)
 		list = append(list, nodepool)
 	}
+	fmt.Println("List ", list)
```
Comment (tahir-duplo) [inline duplocloud/data_source_duplo_gcp_node_pools.go:392]: Please remove these console logs.

File: duplocloud/data_source_duplo_gcp_node_pools.go (hunk 1 above)
Comment (tahir-duplo) [reply duplocloud/data_source_duplo_gcp_node_pools.go:401]: @Duplo-nikv Komal is right here. You just have to return map struct.

Comment (Duplo-nikv) [reply duplocloud/data_source_duplo_gcp_node_pools.go:401]: It was not initializing the values, when used as return so I changed it in this way, which worked

File: duplocloud/data_source_duplo_gcp_node_pools.go (hunk 2 above)
Comment (Duplo-nikv) [reply duplocloud/data_source_duplo_gcp_node_pools.go:392]: done

File: duplocloud/data_source_duplo_gcp_node_pools.go (hunk 1 above)
Comment (Duplo-nikv) [reply duplocloud/data_source_duplo_gcp_node_pools.go]: done

**Pull request merged**